src/ostree/ot-builtin-prune.c

```diff
@@ -103,9 +105,19 @@ traverse_keep_younger_than (OstreeRepo *repo, const char *checksum,
   /* This is the first commit in our loop, which has a ref pointing to it. We
    * don't want to auto-prune it.
    */
-  if (!ostree_repo_traverse_commit_union (repo, checksum, 0, reachable,
-                                          cancellable, error))
-    return FALSE;
+  if (opt_commit_only)
+    {
+      if (!ostree_repo_traverse_commit_only_union (repo, checksum, 0, reachable,
```

Comment:
Having a new "with flags" be the public API simplifies this, because then we can just compute the desired flags, and call the single public API.